src/components/TaskItemContainer.tsx

```diff
@@ -0,0 +1,48 @@
+import { Task } from '../utils/types';
+import { deleteSingleTask, getTasks, updateTask } from '../utils/api';
+import { Dispatch, SetStateAction } from 'react';
+import TaskItemUI from './TaskItemUI';
+
+interface TaskItemContainerProps {
+  task: Task;
+  setTasks: Dispatch<SetStateAction<Task[]>>;
```

Comment:
rather than call it `setTasks`, can we give it a more generic name? For example, `onTaskCompletionChange`; this way, you're not tied to the react set state but rather you're reacting to a event.

Reply:
changed to `updateTaskData`